### DISCUSSION TITLE
Is Windows supposed to be slower?

### DISCUSSION THREAD

**Discussion by Anoromi** [ibhagwan/fzf-lua#1418] [Q&A] [answered]
I'm currently transitioning to fzf-lua from telescope to be able to use my neovim for both windows and linux. I've found that for some reason opening a file through files builtin on windows is slower than on wsl 2 (~500 ms compared to ~100 ms). Is this an expected outcome?

**Comment by ibhagwan** [comment]
Short answer, “yes” (windows cwd.exe), long answer see #1353

**Reply by Anoromi** [reply]
That question centered around openning a window. For me the issue resides in opening a file after fzf does the search. Is it related to needing to end the process after window closes?

**Reply by ibhagwan** [reply]
> That question centered around openning a window. For me the issue resides in opening a file after fzf does the search. Is it related to needing to end the process after window closes?

I would assume so but you can run some tests:
- Where does the delay appeat to be  coming from?
- Is it a delay once you press `enter`  to the time the interface begins to react? Is it possible you have a terminal mapping for enter that makes neovim wait?
- Try other mappings such as `ctrl-s` and `ctr-v` (split and vsplit)
- Is it the same for `files`, `buffers` and `grep` pickers or just for some?

**Reply by Anoromi** [reply]
It occurs in all of the cases except for ctrl-v (because I have it remapped and can't test it). It's highly likely in my opinion that the issue is in closing the process by this point, though I may be wrong. When I use netrw to open files I don't get any of the slowdown.

**Reply by ibhagwan** [reply, marked as answer]
> It occurs in all of the cases except for ctrl-v (because I have it remapped and can't test it). It's highly likely in my opinion that the issue is in closing the process by this point, though I may be wrong. When I use netrw to open files I don't get any of the slowdown.

Netrw doesn't use an external process, windows process management is unfortunately bad, if it's due to the process having to close there's not much I an do about it.

If possible, I highly recommend using WSL on Windows as it doesn't experience all this nonsense.
